app/src/main/java/com/mdg/droiders/samagra/shush/adapters/TimeListAdapter.java

```diff
@@ -0,0 +1,495 @@
+
+        ContentValues values = new ContentValues();
+        values.put(getColumnFromDay(day), false);
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        alarmScheduler.cancelSingleAlarm(day, holder.id);
+    }
+
+    /**
+     * Returns the corresponding column of the day in db.
+     *
+     * @param day The day for which the column is to be returned
+     * @return The column corresponding to the given day
+     */
+    private String getColumnFromDay(int day) {
+        switch (day) {
+            case 0: {
+                return PlacesContract.TimeEntry.COLUMN_MONDAY;
+            }
+            case 1: {
+                return PlacesContract.TimeEntry.COLUMN_TUESDAY;
+            }
+            case 2: {
+                return PlacesContract.TimeEntry.COLUMN_WEDNESDAY;
+            }
+            case 3: {
+                return PlacesContract.TimeEntry.COLUMN_THURSDAY;
+            }
+            case 4: {
+                return PlacesContract.TimeEntry.COLUMN_FRIDAY;
+            }
+            case 5: {
+                return PlacesContract.TimeEntry.COLUMN_SATURDAY;
+            }
+            case 6: {
+                return PlacesContract.TimeEntry.COLUMN_SUNDAY;
+            }
+            default: {
+                return null;
+            }
+        }
+    }
+
+    private void setTimesFromHolder(TimeRowHolder holder, Calendar startTime, Calendar endTime) {
+        String startTimeString = holder.startTime.getText().toString();
+        String endTimeString = holder.endTime.getText().toString();
+        try {
+            startTime.setTime(DISPLAY_DATE_FORMAT.parse(startTimeString));
+            endTime.setTime(DISPLAY_DATE_FORMAT.parse(endTimeString));
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private boolean[] getDayArr(TimeRowHolder holder) {
+        return new boolean[]{
+                holder.monday.isChecked(),
+                holder.tuesday.isChecked(),
+                holder.wednesday.isChecked(),
+                holder.thursday.isChecked(),
+                holder.friday.isChecked(),
+                holder.saturday.isChecked(),
+                holder.sunday.isChecked()
+        };
+    }
+
+    class TimeRowHolder extends RecyclerView.ViewHolder {
+
+        public int id;
+        View itemView;
+        TextView startTime;
+        TextView endTime;
+        Switch enableSwitch;
+        CheckBox monday;
+        CheckBox tuesday;
+        CheckBox wednesday;
+        CheckBox thursday;
+        CheckBox friday;
+        CheckBox saturday;
+        CheckBox sunday;
```

Comment:
An array of Checkboxes would be easier and smaller solution, this way you can iterate over each checkbox in a loop.

Reply:
This is done to avoid arrays in the TimeEntry class so that the database contract is more readable.

Reply:
why not use an array of strings then?